src/components/TodoItem.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TododsContext/TodosContext';
+import { deleteTodo } from '../api/todos';
+import { Loader } from './Loader';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { setErrorMessage, setTodos } = useContext(TodosContext);
+  const [isDeliting, setIsDeliting] = useState(false);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setIsDeliting(true);
+    deleteTodo(todoId)
+      .then(() => setTodos(
+        curentTodos => curentTodos.filter(curtodo => curtodo.id !== todoId),
+      ))
+      .catch((error) => {
+        setErrorMessage('Unable to delete a todo');
+        throw error;
+      })
+      .finally(() => setIsDeliting(false));
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo', {
+        completed: todo.completed,
+      })}
+      key={todo.id}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+        />
+      </label>
+
+      <span data-cy="TodoTitle" className="todo__title">
+        {todo.title}
+      </span>
+
+      <button
+        type="button"
+        className="todo__remove"
+        data-cy="TodoDelete"
+        onClick={() => handleDeleteTodo(todo.id)}
+      >
+        ×
+      </button>
+
+      {(todo.id === 0 || isDeliting) && (
```

Comment:
You are checking if todo.id is 0 to show the loader. However, this check might not be correct because the id could be 0 if it's the first todo. Consider implementing a different mechanism to show the loader when waiting for the API response.